packages/CodingStandard/src/Fixer/DependencyInjection/InjectToConstructorInjectionFixer.php

```diff
@@ -0,0 +1,239 @@
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_FUNCTION)) {
+                    $namePosition = $tokens->getNextNonWhitespace($i);
+                    $methodNameToken = $tokens[$namePosition];
+                    if ($methodNameToken->getContent() === '__construct') {
+                        $constructMethodPosition = $i;
+                        break;
+                    }
+                }
+            }
+
+            // A. has a constructor?
+            if (is_int($constructMethodPosition)) { // "function" token
+                $this->addPropertyToConstructor($tokens, $propertyType, $propertyName, $constructMethodPosition);
+            } else {
+                // B. doesn't have a constructor?
+                $this->addConstructorMethod($tokens, $propertyType, $propertyName);
+            }
+
+            // run again with new tokens; @todo: can this be done any better to notice new __construct method added
+            // by these tokens?
+            $this->fix($file, $tokens);
+            break;
+        }
+    }
+
+    public function isRisky(): bool
+    {
+        return false;
+    }
+
+    public function getName(): string
+    {
+        return self::class;
+    }
+
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    public function supports(SplFileInfo $file): bool
+    {
+        return true;
+    }
+
+    private function addConstructorMethod(Tokens $tokens, string $propertyType, string $propertyName): void
+    {
+        $constructorPosition = $this->getConstructorPosition($tokens);
+        $constructorTokens = $this->createConstructorWithPropertyCodeInTokens($propertyType, $propertyName);
+
+        $tokens->insertAt($constructorPosition, $constructorTokens);
+    }
+
+    private function getConstructorPosition(Tokens $tokens): int
+    {
+        // 1. after last property
+        for ($index = count($tokens) - 1; $index > 1; --$index) {
+            $token = $tokens[$index];
+            if ($token->isGivenKind(T_VARIABLE)) {
+                $propertyEndSemicolonPosition = $tokens->getNextTokenOfKind($index, [';']);
+
+                return $propertyEndSemicolonPosition + 1;
+            }
+        }
+
+        // 2. before first method
+        foreach ($tokens as $index => $token) {
+            if ($token->isGivenKind(T_FUNCTION)) {
+                $methodStartPosition = $tokens->getPrevTokenOfKind(
+                    $index,
+                    [T_PUBLIC, T_PRIVATE, T_PROTECTED, T_DOC_COMMENT]
+                );
+
+                return $methodStartPosition - 1;
+            }
+        }
+    }
+
+    private function createConstructorWithPropertyCodeInTokens(string $propertyType, string $propertyName): Tokens
+    {
+        $indentedConstructorCode = $this->createConstructorWithPropertyCodeInString($propertyType, $propertyName);
+        $constructorTokens = Tokens::fromCode(sprintf('<?php class SomeClass {
```

Comment:
while effective this code might become fragile, I would suggest making the tokens yourself here like you did here; https://github.com/Symplify/Symplify/pull/236/files#diff-0c7942472a912c2c38e42b9bc5edb5c7R213